pkgs/by-name/an/antimatter-dimensions/package.nix

```
@@ -0,0 +1,76 @@
+{ buildNpmPackage
+, electron
+, fetchFromGitHub
+, lib
+, makeDesktopItem
+, unstableGitUpdater
+, writeScriptBin
+, electronAppName ? "Antimatter Dimensions"
+}:
+
+let
+ dummyElectronApp = ./app.js;
+in
+buildNpmPackage rec {
+  pname = "antimatter-dimensions";
+  version = "0-unstable-2024-05-11";
+  src = fetchFromGitHub {
+    owner = "IvarK";
+    repo = "AntimatterDimensionsSourceCode";
+    rev = "b3a254af60207a03d04473bb81726e921f5b2c61";
+    hash = "sha256-+G9mNilt5Ewja5P+Bt312EcCJknMu7FOMn5b4FseAyQ=";
+  };
+  nativeBuildInputs = [
+    # build script calls git to get git hash, message and author
+    # since fetchFromGitHub doesn't provide this information
+    # create a dummy git
+    (writeScriptBin "git" ''
+      echo "unknown"
+    '')
+  ];
```

```suggestion
  src = fetchFromGitHub {
    owner = "IvarK";
    repo = "AntimatterDimensionsSourceCode";
    rev = "b3a254af60207a03d04473bb81726e921f5b2c61";
    hash = "sha256-+G9mNilt5Ewja5P+Bt312EcCJknMu7FOMn5b4FseAyQ=";
    leaveDotGit = true;
  };
  nativeBuildInputs = [
    git
  ];
```

Obviously git has to also be added to the inputs.

---

using `leaveDotGit` will make the output non-deterministic, since obviously .git can change easily. You'll get random hash mismatchs. If you really need the commit hash, message, id, you can [design an updateScript](https://github.com/NixOS/nixpkgs/commit/7b136032a56ca0bf1eac8d32bbf298a8dafdf976), that'll generate a structured output (json) and use it.

---

> using leaveDotGit will make the output non-deterministic

How so? The only stuff that uses `.git` is https://github.com/IvarK/AntimatterDimensionsSourceCode/blob/master/build/post-build.js#L9-L13, which depend entirely on the `HEAD` commit, which is fixed in `fetchFromGitHub`.

---

See https://github.com/NixOS/nixpkgs/issues/8567.

---

Fair enough. Mark this as resolved too.

---

@amozeo please revert this change, using `leaveDotGit` is not desireable.